libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -862,4 +880,26 @@ class MapboxRouteLineView(options: MapboxRouteLineOptions) {
                 }
             }
     }
+
+    private fun resetLayers(style: Style) {
+        sourceToFeatureMap.clear()
+        sourceToFeatureMap[MapboxRouteLineUtils.layerGroup1SourceKey] = RouteLineFeatureId(null)
+        sourceToFeatureMap[MapboxRouteLineUtils.layerGroup2SourceKey] = RouteLineFeatureId(null)
+        sourceToFeatureMap[MapboxRouteLineUtils.layerGroup3SourceKey] = RouteLineFeatureId(null)
+        primaryRouteLineLayerGroup = setOf()
+        listOf(
+            RouteLayerConstants.LAYER_GROUP_1_SOURCE_ID,
+            RouteLayerConstants.LAYER_GROUP_2_SOURCE_ID,
+            RouteLayerConstants.LAYER_GROUP_3_SOURCE_ID,
+            RouteLayerConstants.WAYPOINT_SOURCE_ID
+        ).forEach {
+            updateSource(
+                style,
+                it,
+                FeatureCollection.fromFeatures(listOf())
+            )
+        }
```

Comment:
Do we need this? Are you still able to reproduce the issue where clearing sources is needed to re-apply layer properties?

Reply:
Yes. if the sources aren't updated with an empty FeatureCollection the lines appear on the map even though there are calls to remove the sources and add them back. I don't know if this is a bug in maps or it is by design with a remove source call immediately followed by an add source call with the same source ID.   However the lines lack the correct coloring because the gradient data isn't preserved and the view class doesn't have it.  

In my opinion an initialization of the layers whether implicit or explicit should always result in the same map state with regards to the route line layers.  This implementation achieves that.